Converted projects to xproj/project.json for multitargeting ...

prof79 opened the pull request:
(dotnet pack) all current .NET versions >= 4.0 including .NET Standard (aka .NET Core).
Tested/built on VS 2015 Community Update 3 w/ DotNetCore.1.0.1-VS2015Tools.Preview2.0.3.exe installed.
Just run "dotnet pack -c Release" from a shell in the project directory and you will get a multiplatform all-in-one .nupkg.
Even though xproj/project.json will change this seems to be the simplest and most hassle-free way until VS 2017 matures. There will be a "dotnet migrate" migration path in new VS.
I'm just a hobbyist programmer so I'm still unsure about how many framework combinations are really necessary in the project files; MSTest doesn't like to be built on ,NETStandard but 4.5.x works and passes so I guess there is not much of a concern.

stevehansen commented:
Awesome, thanks for doing this.

Before we can merge though, having the old .sln/.csproj files will cause appveyor to skip building because it doesn't know which one to build, might be better to just delete them?
I don't think we are supposed to check in the .lock.json files, I would also remove those and add them to the .gitignore file.

prof79 commented:
"As you wish" ;-)
Cleaned up the old project files, kept em first as a good measure and for reference in regard to any properties I might have missed.

project.lock.json makes me shrug, I haven't fully understood the implications yet. Maybe you can make more sense of this: [http://blog.brunomlopes.com/2015/07/should-i-put-projectlockjson-in-source.html](url)

Anyway, followed your suggestion to add to .gitignore and removed them from the branch.
